`Web docs` - Moving table based content to a new section. (#3157)
hashicorp/design-system: website/docs/content/component/tables/index.md

---
title: Tables
caption: Writing guidance for tabular data.
description: Writing guidance for tabular data.
previewImage: assets/illustrations/content/writing-guidelines.jpg
related:
  - components/table/table
  - components/table/advanced-table
---

> This documentation provides best practices for writing tabular content, such as with the [Table](/components/table/table) and [Advanced Table](/components/table/advanced-table) components.

Reviewer: Nitpick 
```suggestion
This documentation provides best practices for writing tabular content in [Table](/components/table/table) and [Advanced Table](/components/table/advanced-table) components.
```

## General recommendations

While we are not prescriptive about what goes into table cells, there are some best practices to consider:

> - Keep data within columns to one data type. Sorting on multiple data types is technically difficult and makes content confusing for assistive technologies.

Reviewer: Nitpick: wording

```suggestion
- Keep data within columns to one data type. Sorting on multiple data types adds technical complexity and makes content confusing for assistive technologies.
```

![Icon within a table without a label](/assets/content/component/tables/icon-without-label.png)
!!!

!!! Dont

Don’t use an icon to indicate the status of an object, row, or resource. Instead, use a [Badge](/components/badge).

![Icons being used for status](/assets/content/component/tables/icon-status.png)
!!!

### Service icons

Use service icons within a cell to communicate the source or provider of a service.

![Service icon within a table](/assets/content/component/tables/with-service-icon.png)

### Grouping

Use icons to communicate commonalities between values or that a value is part of a larger object or hierarchical structure.

![Icons used for grouping in a table](/assets/content/component/tables/icon-grouping.png)

### Product branding

Use icons to communicate that a specific item is a HashiCorp product or resource.

![Icon product branding](/assets/content/component/tables/icon-product-branding.png)

!!! Dont

Don’t use an [Icon Tile](/components/icon-tile) in place of an icon within a table cell.

![Icon Tile within a table cell](/assets/content/component/tables/icon-tile-product-branding.png)
!!!

### Leading vs. trailing icons

> In general, we recommend using **leading icons** because the text following the icon will remain aligned and thus be easier for the user to visually scan.

Reviewer: Nitpick: wording, brevity

```suggestion
We recommend using **leading icons** so the text following the icon remains aligned down the column for better scannability.
```

- `Neutral` for [null and empty values](#null-values), e.g., "None", "No status", etc.

<Doc::Layout @direction="horizontal" @spacing="16px">
  <Hds::Badge @text="Active" @color="success" @icon="check" @type="outlined" />
  <Hds::Badge @text="Degraded" @color="warning" @icon="alert-triangle" @type="outlined" />
  <Hds::Badge @text="Failed" @color="critical" @icon="x" @type="outlined" />
  <Hds::Badge @text="Starting up" @color="highlight" @icon="loading" @type="outlined" />
  <Hds::Badge @text="No status" @color="neutral" @type="outlined" />
</Doc::Layout>

### Badge icon usage

Use logical icons when communicating status in a Badge. Some common examples when paired with Badge `color` include:

- `check` for positive communication.
- `alert-triangle` for cautionary communication.
- `x` for negative communication and errors.
- `loading` for communicating a dynamic value or status and when using `color=highlight`.

!!! Do

In the case of a [null or empty](#null-values) value, use the text-only variant of the Badge.

![Null value represented by a badge within a table](/assets/content/component/tables/badge-table-null-value.png)
!!!

### Badge consistency

> Ensure that Badge usage within tables is consistent across features and within an application holistically. For example, if communicating that a record is "active", use the same combination of text, icon, and color in each instance. This consistency in communication and visual language can make complex information easier to understand and quicker to parse.

Reviewer: Nitpick: I feel like this could be moved under the H2 and abbreviated slightly rather than take its own section
